### PR TITLE
Airbubble cooling update

### PR DESCRIPTION
**Pull request by Sindorman**
Airbubble now has cooling unit capability:

- Airbubble cools down mobs inside with same efficiency as cooling unit and 2 time less charge consumption.

- Require power cell to be attached in order to work.

- When examined it tells what kind of cell is attached and what is the charge of it.

### CODE REVIEW

**Comment by Sindorman** [comment]
!review

File: code/game/objects/items/airbubble.dm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Regular airbubble - folded
 /obj/item/airbubble
 	name = "air bubble"
-	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. Has an integrated cooling unit to preserve stable temperature inside, require power cell to operate."
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
	desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate."
```

File: code/game/objects/items/airbubble.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@
 // Deployed bubble
 /obj/structure/closet/airbubble
 	name = "air bubble"
-	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. Has an integrated cooling unit to preserve stable temperature inside, require power cell to operate."
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
	desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate."
```

File: code/game/objects/items/airbubble.dm [hunk 3]
```diff
@@ -85,6 +89,10 @@
 		to_chat(user, "<span class='notice'>\The [src] has [internal_tank] attached, that displays [round(internal_tank.air_contents.return_pressure() ? internal_tank.air_contents.return_pressure() : 0)] KPa.</span>")
 	else
 		to_chat(user, "<span class='notice'>\The [src] has no tank attached.</span>")
+	if (cell)
+		to_chat(user, "\The [src] has [cell] attached, and the charge meter reads [round(cell.percent())]%.")
+	else
+		to_chat(user, "<span class='warning'> It doesn't have a power cell installed.</span>")
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
		to_chat(user, "<span class='warning'>[src] has no power cell installed.</span>")
```
To conform with the other messages

File: code/game/objects/items/airbubble.dm [hunk 4]
```diff
@@ -85,6 +89,10 @@
 		to_chat(user, "<span class='notice'>\The [src] has [internal_tank] attached, that displays [round(internal_tank.air_contents.return_pressure() ? internal_tank.air_contents.return_pressure() : 0)] KPa.</span>")
 	else
 		to_chat(user, "<span class='notice'>\The [src] has no tank attached.</span>")
+	if (cell)
+		to_chat(user, "\The [src] has [cell] attached, and the charge meter reads [round(cell.percent())]%.")
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
		to_chat(user, "\The [src] has [cell] attached, the charge meter reads [round(cell.percent())]%.")
```

File: code/game/objects/items/airbubble.dm [hunk 5]
```diff
@@ -183,6 +191,9 @@
 		if(!ishuman(usr))	return
 		if(opened)	return 0
 		if(contents.len > 1)	return 0
+		if(cell)
+			to_chat(usr, "<span class='warning'>[src] still has [cell] attached to it, please remove it before folding \the [src].</span>")
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
			to_chat(usr, "<span class='warning'>[src] can not be folded with [cell] attached to it.</span>")
```

File: code/game/objects/items/airbubble.dm [hunk 6]
```diff
@@ -207,10 +218,10 @@
 			internal_tank = null
 		bag.w_class = ITEMSIZE_LARGE
 
-		bag.desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+		bag.desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. Has an integrated cooling unit to preserve stable temperature inside, require power cell to operate."
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
		bag.desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate.
```

File: code/game/objects/items/airbubble.dm [hunk 7]
```diff
@@ -533,13 +595,13 @@
 // Syndicate airbubble
 /obj/item/airbubble/syndie
 	name = "air bubble"
-	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. This does not seem like a regular color scheme"
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. Has an integrated cooling unit to preserve stable temperature inside, require power cell to operate. This does not seem like a regular color scheme"
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
	desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate. This does not seem like a regular color scheme"
```

File: code/game/objects/items/airbubble.dm [hunk 8]
```diff
@@ -533,13 +595,13 @@
...
 	icon_state = "airbubble_syndie_fact_folded"
 	syndie = TRUE
 
 /obj/structure/closet/airbubble/syndie
 	name = "air bubble"
-	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. This does not seem like a regular color scheme"
+	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. Has an integrated cooling unit to preserve stable temperature inside, require power cell to operate. This does not seem like a regular color scheme"
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
	desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate. This does not seem like a regular color scheme"
```

File: html/changelogs/Sindorman-airbubble.yml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - rscadd: "Aribbubble now has intergrated cooling unit. It require a power cell. to be attached either externally or internally by clicking on the bubble with power cell. Bubble cools down as fast as suit cooling unit and cosumes 2 times slower charge then suit cooling unit."
```
Comment (Arrow768) [inline html/changelogs/Sindorman-airbubble.yml]:
```suggestion
  - rscadd: "Aribbubbles now have a intergrated cooling unit. It require a power cell to be attached either externally or internally by clicking on the bubble with power cell. The bubble cools down as fast as suit cooling unit and consumes 2 times less charge than a suit cooling unit."
```

File: html/changelogs/Sindorman-airbubble.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  - tweak: "Airbubble now shows what kind of cell is attached to it and what is the charge of it when examined."
```
Comment (Arrow768) [inline html/changelogs/Sindorman-airbubble.yml]:
```suggestion
  - tweak: "Airbubbles now show the attached cell type and its charge when examined."
```

File: code/game/objects/items/airbubble.dm [hunk 11]
```diff
@@ -207,10 +218,10 @@
 			internal_tank = null
 		bag.w_class = ITEMSIZE_LARGE
 
-		bag.desc = "Special air bubble designed to protect people inside of it from decompressed enviroments."
+		bag.desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate.
```
Comment (Arrow768) [inline code/game/objects/items/airbubble.dm]:
```suggestion
		bag.desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate."
```
Forgot the closing "

File: code/game/objects/items/airbubble.dm [hunk 12]
```diff
@@ -533,13 +595,13 @@
...
 	name = "air bubble"
-	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. This does not seem like a regular color scheme"
+	desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate. This does not seem like a regular color scheme"
 	icon_state = "airbubble_syndie_fact_folded"
 	syndie = TRUE
 
 /obj/structure/closet/airbubble/syndie
 	name = "air bubble"
-	desc = "Special air bubble designed to protect people inside of it from decompressed enviroments. This does not seem like a regular color scheme"
+	desc = "Special air bubble designed to protect people inside of it from decompressed environments. Has an integrated cooling unit to preserve a stable temperature inside. Requires a power cell to operate. This does not seem like a regular color scheme"
```
Comment (BurgerLUA) [inline code/game/objects/items/airbubble.dm]: Forgot a period at the end of this and the non-closet variant.

File: html/changelogs/Sindorman-airbubble.yml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+#   imageadd
+#   imagedel
+#   maptweak
+#   spellcheck (typo fixes)
+#   experiment
+#   balance
+#################################
+
+# Your name.
+author: N3X15
```
Comment (BurgerLUA) [inline html/changelogs/Sindorman-airbubble.yml]: this isn't you

**Comment by Sindorman** [comment]
apparently I screwed up during rebase

File: code/game/objects/items/airbubble.dm [hunk 14]
```diff
@@ -85,6 +89,10 @@
 		to_chat(user, "<span class='notice'>\The [src] has [internal_tank] attached, that displays [round(internal_tank.air_contents.return_pressure() ? internal_tank.air_contents.return_pressure() : 0)] KPa.</span>")
 	else
 		to_chat(user, "<span class='notice'>\The [src] has no tank attached.</span>")
+	if (cell)
+		to_chat(user, "\The [src] has [cell] attached, the charge meter reads [round(cell.percent())]%.")
+	else
+		to_chat(user, "<span class='warning'>[src] has no power cell installed.</span>")
```
Comment (BurgerLUA) [inline code/game/objects/items/airbubble.dm]: Should be "/The [src] has no power cell installed.

File: code/game/objects/items/airbubble.dm [hunk 15]
```diff
@@ -183,6 +191,9 @@
 		if(!ishuman(usr))	return
 		if(opened)	return 0
 		if(contents.len > 1)	return 0
+		if(cell)
+			to_chat(usr, "<span class='warning'>[src] can not be folded with [cell] attached to it.</span>")
```
Comment (BurgerLUA) [inline code/game/objects/items/airbubble.dm]: Should be "/The [src] can not be folder with /the [src] attached to it."

File: code/game/objects/items/airbubble.dm [hunk 16]
```diff
@@ -333,6 +344,29 @@
...
 		to_chat(usr, "<span class='warning'>[src] has no tank.</span>")
 
+// Remove tank from bubble
+/obj/structure/closet/airbubble/verb/take_cell()
+	set src in oview(1)
+	set category = "Object"
+	set name = "Detach Power Cell"
+	if(!isnull(cell))
+		usr.visible_message(
+		"<span class='warning'>[usr] is removing [cell] from [src].</span>",
```
Comment (BurgerLUA) [inline code/game/objects/items/airbubble.dm]: Missing /the in front of all the []s

File: code/game/objects/items/airbubble.dm [hunk 17]
```diff
@@ -333,6 +344,29 @@
...
+		"<span class='notice'>You are removing [cell] from [src].</span>"
```
Comment (BurgerLUA) [inline code/game/objects/items/airbubble.dm]: Missing /the

File: code/game/objects/items/airbubble.dm (hunk 14 above)
Comment (skull132) [reply code/game/objects/items/airbubble.dm]: Unnecessary: the newer (at this point, by years) standard has names properly prepended with `\proper` or `\improper`, which will be utilized when they're put into a string directly. `\the` is required when you use something like `[src.name]`.

File: code/game/objects/items/airbubble.dm [hunk 18]
```diff
@@ -333,6 +344,29 @@
 	else
 		to_chat(usr, "<span class='warning'>[src] has no tank.</span>")
 
+// Remove tank from bubble
+/obj/structure/closet/airbubble/verb/take_cell()
+	set src in oview(1)
+	set category = "Object"
+	set name = "Detach Power Cell"
```
Comment (skull132) [inline code/game/objects/items/airbubble.dm]: Unguarded verb. Literally any ghost or mouse can use this. Use the `CanUse()` proc to validate the user.

**Pull request merged**